persistence/sql/persister_consent.go

```diff
@@ -216,11 +216,18 @@ func (p *Persister) VerifyAndInvalidateConsentRequest(ctx context.Context, verif
 
 func (p *Persister) HandleLoginRequest(ctx context.Context, challenge string, r *consent.HandledLoginRequest) (lr *consent.LoginRequest, err error) {
 	return lr, p.transaction(ctx, func(ctx context.Context, c *pop.Connection) error {
-		err := c.Create(r)
+		found, err := c.Where("challenge = ?", challenge).Exists(&consent.HandledLoginRequest{})
 		if err != nil {
 			return sqlcon.HandleError(err)
 		}
 
+		if !found {
```

Comment:
This would only work in the case of an exact retry. It will be confusing for developers who expect the second call to override the first one, which is why we wanted to return 429 with an URL the developers can use for next action.

Reply:
Looks like consent is supported already https://github.com/ory/hydra/pull/2914#issuecomment-1008902061. Do you see the need for further work to ensure that enabling this at all is a secure solution @aeneasr? I.e. is it not realistic to get a change like this merged as more of a quick fix?